Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -77,50 +74,33 @@ float TrackletTransformer::calculateZ(int padrow)
   return rowPos - rowSize / 2. - middleRowPos;
 }
 
-float TrackletTransformer::calculateDy(int slope, double oldLorentzAngle, double lorentzAngle, double driftVRatio)
+float TrackletTransformer::calculateDy(int slope, double lorentzAngle, double driftVRatio)
 {
   double padWidth = mPadPlane->getWidthIPad();
 
   // temporary dummy value in cm/microsecond
   float vDrift = 1.5464f;
-  float driftHeight = mGeo->cdrHght();
 
-  int dYsigned = 0;
+  int slopeSigned = 0;
   if (slope & (1 << (NBITSTRKLSLOPE - 1))) {
-    dYsigned = -((~(slope - 1)) & ((1 << NBITSTRKLSLOPE) - 1));
+    slopeSigned = -((~(slope - 1)) & ((1 << NBITSTRKLSLOPE) - 1));
   } else {
-    dYsigned = slope & ((1 << NBITSTRKLSLOPE) - 1);
+    slopeSigned = slope & ((1 << NBITSTRKLSLOPE) - 1);
   }
+
   // dy = slope * nTimeBins * padWidth * GRANULARITYTRKLSLOPE;
   // nTimeBins should be number of timebins in drift region. 1 timebin is 100 nanosecond
-  double rawDy = dYsigned * ((driftHeight / vDrift) * 10.) * padWidth * GRANULARITYTRKLSLOPE;
-
-  // driftDistance = 3.35
-  float driftDistance = mGeo->cdrHght() + mGeo->camHght();
-
-  float cmSlope = rawDy / driftDistance;
-
-  double calibratedDy = rawDy - (TMath::Tan(lorentzAngle) * driftDistance);
-  calibratedDy += (TMath::Tan(oldLorentzAngle) * driftDistance * driftVRatio) + cmSlope * (driftDistance * (1 - driftVRatio));
-
-  // ALTERNATIVE METHOD
-
-  // double x_anode_hit = driftDistance*driftVRatio/cmSlope;
-  // double y_anode_hit = driftDistance*driftVRatio;
-
-  // double x_Lorentz_drift_hit = TMath::Tan(oldLorentzAngle)*driftDistance*driftVRatio - TMath::Tan(lorentzAngle)*driftDistance;
-  // double y_Lorentz_drift_hit = driftDistance*driftVRatio - driftDistance;
-
-  // double Delta_x_Lorentz_drift_hit = x_anode_hit - x_Lorentz_drift_hit;
-  // double Delta_y_Lorentz_drift_hit = y_anode_hit - y_Lorentz_drift_hit;
-  // double impact_angle_rec = TMath::ATan2(Delta_y_Lorentz_drift_hit,Delta_x_Lorentz_drift_hit);
+  double rawDy = slopeSigned * ((mXAnode / vDrift) * 10.) * padWidth * GRANULARITYTRKLSLOPE;
```

Comment:
This should be `mXCathode` instead of `mXAnode` I think

Reply:
I can't recall if you were in the meeting when we talked about this, but we decided to go with the anode height of 3.35 cm for now. This will be consistent with what is used in the calibration code. However, we will likely revisit this issue in the future since there is still disagreement about 3.35 cm being the most sensible value here, but the change needs to happen in calibration code as well.

Reply:
Well, but if you change it here then this will affect the tracking. Couldn't you change this to be a correction factor for `calibratedDy`? For me `rawDy` should be the deflection that is calculated in the TRAP and there the drift length of 3cm is used and not 3.35 cm.

Reply:
Should be resolved now

Reply:
Looks good, thanks